sharktank/sharktank/models/llama/llama.py

```diff
@@ -275,27 +284,46 @@ def __init__(
         head_dim: int,
         head_count_kv: int,
         rms_epsilon: float,
+        hf: bool = False,
     ):
         super().__init__(theta)
-        self.add_module(
-            "attn_norm", RMSNormLayer(theta("attn_norm"), epsilon=rms_epsilon)
-        )
-        self.add_module("attn_q", LinearLayer(theta("attn_q")))
-        self.add_module("attn_k", LinearLayer(theta("attn_k")))
-        self.add_module("attn_v", LinearLayer(theta("attn_v")))
-        self.add_module("attn_output", LinearLayer(theta("attn_output")))
-        self.add_module(
-            "ffn_norm", RMSNormLayer(theta("ffn_norm"), epsilon=rms_epsilon)
-        )
-        self.add_module("ffn_gate", LinearLayer(theta("ffn_gate")))
-        self.add_module("ffn_up", LinearLayer(theta("ffn_up")))
-        self.add_module("ffn_down", LinearLayer(theta("ffn_down")))
+        if hf:
+            self.add_module(
+                "attn_norm", RMSNormLayer(theta("input_layernorm"), epsilon=rms_epsilon)
+            )
+            self.add_module("attn_q", LinearLayer(theta("self_attn.q_proj")))
```

Comment:
Do we need this block of code to rename the parameters? Since in the test we are specifying the paramter names anyways, couldn't we just use the old names?

Reply:
I think this is because sharktank is based on gguf instead of huggingface, so the name mappings are different and need to be specified when using gguf weights vs. huggingface weights.

Reply:
But looks like the huggingface "weight" is just being loaded here, where we specify through string what the weights are, but technically, we can load it using the old string/params name no? https://github.com/nod-ai/sharktank/pull/120/files#diff-3605352cc960ff4b5e59079e25b9f8db20a8bd83646b1788d73e6d2d14bd4ec3R108-R131

Reply:
Decided to use original string/params names

Reply:
awesome!